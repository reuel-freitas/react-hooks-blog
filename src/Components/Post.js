export const Post = ({ post }) => {
    return (
        <div style={{ backgroundColor: '#6e6e6e6e', padding: 10, borderRadius: 7 }}>
            <img src={post.cover} alt="Imagem do post" style={{ maxWidth: '100%', borderRadius: 7 }} />
            <h3 style={{ color: '#eee', textAlign: 'left' }}>{post.title}</h3>
            <p style={{ color: '#eee', textAlign: 'left' }}>{post.body}</p>
        </div>
    )
}