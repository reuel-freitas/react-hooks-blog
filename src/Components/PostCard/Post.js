export const Post = ({ cover, title, body }) => {
    return (
        <div style={{ backgroundColor: '#6e6e6e6e', padding: 10, borderRadius: 7 }}>
            <img src={cover} alt={title} style={{ maxWidth: '100%', borderRadius: 7 }} />
            <h3 style={{ color: '#eee', textAlign: 'left' }}>{title}</h3>
            <p style={{ color: '#eee', textAlign: 'left' }}>{body}</p>
        </div>
    )
}