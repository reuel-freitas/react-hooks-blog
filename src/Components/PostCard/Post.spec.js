import { render, screen } from "@testing-library/react"
import { postPropsMock } from "./mock"
import { Post } from "./Post"

const props = postPropsMock

describe('<Post />', () => {
    it('should render PostCard correctly', () => {
        render(<Post {...props} />)

        expect(screen.getByRole('img', { name: props.title }))
            .toHaveAttribute('src', props.cover)

        expect(screen.getByRole('heading', { name: props.title }))
            .toBeInTheDocument()

        expect(screen.getByText(props.body))
            .toBeInTheDocument()

        expect.assertions(3)
    })

    it('should match snapshot', () => {
        const { container } = render(<Post {...props} />)
        expect(container.firstChild).toMatchSnapshot()
    })
})