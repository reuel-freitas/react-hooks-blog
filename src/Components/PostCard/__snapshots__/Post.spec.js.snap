// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Post /> should match snapshot 1`] = `
<div
  style="background-color: rgba(110, 110, 110, 0.431); padding: 10px; border-radius: 7px;"
>
  <img
    alt="title 1"
    src="img/img.png"
    style="max-width: 100%; border-radius: 7px;"
  />
  <h3
    style="color: rgb(238, 238, 238); text-align: left;"
  >
    title 1
  </h3>
  <p
    style="color: rgb(238, 238, 238); text-align: left;"
  >
    body 1
  </p>
</div>
`;
